### PR TITLE
weekly interest label

### PR DESCRIPTION
**Pull request by ankur-singh-dehaat**

### CODE REVIEW

File: lib-ledger/src/main/java/lib/dehaat/ledger/entities/revamp/transaction/TransactionEntityV2.kt [hunk 1]
```diff
@@ -15,5 +15,8 @@ data class TransactionEntityV2(
 	val sourceNo: String?,
 	val type: String,
 	val unrealizedPayment: Boolean?,
-	val isInterestSubVented: Boolean?
+	val isInterestSubVented: Boolean?,
+    val fromDate: Long?,
```
Comment (shivam21) [inline lib-ledger/src/main/java/lib/dehaat/ledger/entities/revamp/transaction/TransactionEntityV2.kt:19]: format

File: lib-ledger/src/main/java/lib/dehaat/ledger/presentation/ledger/revamp/state/transactions/ui/TransactionsScreen.kt [hunk 2]
```diff
@@ -47,13 +56,37 @@ fun TransactionsScreen(
...
+            if (uiState.showWeeklyInterestDecreasingLabel) {
+                Text(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .background(Color.White)
+                        .padding(horizontal = 20.dp, vertical = 12.dp)
+                        .background(
+                            brush = Brush.horizontalGradient(
+                                colors = listOf(
+                                    Color(0xff3985BF),
```
Comment (shivam21) [inline lib-ledger/src/main/java/lib/dehaat/ledger/presentation/ledger/revamp/state/transactions/ui/TransactionsScreen.kt:77]: move colors to Colors.kt

File: lib-ledger/src/main/java/lib/dehaat/ledger/presentation/mapper/LedgerViewDataMapper.kt [hunk 3]
```diff
@@ -126,7 +127,10 @@ class LedgerViewDataMapper @Inject constructor() {
 			source = it.source,
 			sourceNo = it.sourceNo,
 			type = it.type,
-			unrealizedPayment = it.unrealizedPayment
+			unrealizedPayment = it.unrealizedPayment,
+            fromDate = it.fromDate?.toDateMonthYear(),
```
Comment (shivam21) [inline lib-ledger/src/main/java/lib/dehaat/ledger/presentation/mapper/LedgerViewDataMapper.kt:131]: format

**Review by shivam21 — CHANGES_REQUESTED** [review]

**Review by lovemhjn — APPROVED** [review]
lgtm

**Pull request merged**